src/middleware.rs

```diff
@@ -164,3 +164,79 @@ where
         (self)(request, next)
     }
 }
+
+#[cfg(feature = "digest-auth")]
+pub mod digest {
+    use super::*;
+    use crate::{Request, Response};
+    use digest_auth::{AuthContext, WwwAuthenticateHeader};
+    use std::{borrow::Cow, str::FromStr};
+
+    /// Provides simple digest authentication powered by the `digest_auth` crate.
+    ///
+    /// Any 401 response handled by this middleware is retried once with the
+    /// credentials provided on construction. If authentication fails or the middleware
+    /// is unable to generate an answer to the server challenge (such as a different authentication
+    /// scheme or a malformed challenge) the response is silently passed on to the rest
+    /// of the middleware chain.
```

Comment:
```suggestion
    /// Requests that receive a HTTP 401 response are retried once by this middleware with the
    /// credentials provided on construction. The retry only happens under these conditions:
    /// - there is no prior "authorization" header on the request set by the caller or other
    ///   middleware, and
    /// - the server provides HTTP Digest auth challenge in the "www-authenticate" header.
    ///
    /// In other cases, this middle ware acts as a no-op forwarder of requests and responses.
```

^^^ my attempt to specify exact cases in which this middleware kicks in.